extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/runtime/JpaOperations.java

```diff
@@ -155,6 +155,29 @@ private static String createCountQuery(Class<?> entityClass, String query, int p
         return "SELECT COUNT(*) FROM " + getEntityName(entityClass) + " WHERE " + query;
     }
 
+    private static String createUpdateQuery(Class<?> entityClass, String query, int paramCount) {
+        if (query == null) {
+            return "UPDATE FROM " + getEntityName(entityClass);
+        }
+
+        String trimmed = query.trim();
+        if (trimmed.isEmpty()) {
+            return "UPDATE FROM " + getEntityName(entityClass);
```

Comment:
I don't think those two first cases mean anything. I doubt they will be accepted by Hibernate ORM so we should throw here, or possibly there could be a case for making the caller return a `0` update count for empty updates, to be like `count` or `find` and find a possible meaning.

Reply:
You are right, both are invalid. I thought about throwing an exception, but I don't know what kind of exception it should be. org.hibernate.QueryException("Query string cannot be empty")?

Reply:
There is a newly added `io.quarkus.panache.common.exception.PanacheQueryException` maybe use this one as it is not an Hibernate exception but a Panache exception.

Reply:
Done